algebird-core/src/main/scala/com/twitter/algebird/QTree.scala

```diff
@@ -108,32 +152,45 @@ class QTreeSemigroup[A](k: Int)(implicit val underlyingMonoid: Monoid[A]) extend
   }
 }
 
-case class QTree[A](
+case class QTree[@specialized(Int, Long, Float, Double, Unit) A] private[algebird] (
```

Comment:
it's a bummer to lose the case class matching, but don't we need to add:

```scala
  def offset
  def level
  def count
  def sum
```
or we are breaking the API?

For instance, Zak has some code to serialize QTrees, I'm not sure if that code can work without access to those values (pretty sure it can't).

I don't know how private interacts with case classes actually.